src/cgame/cg_ents.cpp

```diff
@@ -1317,39 +1316,43 @@ void CG_AddPacketEntities()
 	// lerp the non-predicted value for lightning gun origins
 	CG_CalcEntityLerpPositions( &cg_entities[ cg.snap->ps.clientNum ] );
 
-	for ( unsigned num = 0; num < MAX_GENTITIES; num++ )
-	{
-		cg_entities[ num ].valid = false;
-	}
+	bool done[ MAX_GENTITIES ] = {};
 
 	// add each entity sent over by the server
-	for ( unsigned num = 0; num < cg.snap->entities.size(); num++ )
+	for ( unsigned n = 0; n < cg.snap->entities.size(); n++ )
```

Comment:
You can use `for(const entityState_t &ent : cg.snap->entities)` for efficient loop iteration

Reply:
Done.